pkg/errors/broadcasting.go

```diff
@@ -0,0 +1,213 @@
+	CreateAction BroadcastOperation = "createAction"
+
+	// ProcessAction represents an action to process an existing wallet action.
+	ProcessAction BroadcastOperation = "processAction"
+)
+
+// BroadcastingError represents an error that occurred during transaction broadcasting
+type BroadcastingError struct {
+	Err             error
+	TxID            string
+	Reference       string
+	SendWithResults []wdk.SendWithResult
+	ReviewResults   []wdk.ReviewActionResult
+	ServiceErrors   map[string]error
+	Operation       BroadcastOperation
+	Tx              []byte
+	NoSendChange    []wdk.OutPoint
+}
+
+// Error implements the error interface for BroadcastingError
+func (e *BroadcastingError) Error() string {
+	if e.Err == nil {
+		return fmt.Sprintf("broadcasting failed during %s for txID %s (reference: %s)",
+			e.Operation, e.TxID, e.Reference)
+	}
+	return fmt.Sprintf("broadcasting failed during %s for txID %s (reference: %s): %v",
+		e.Operation, e.TxID, e.Reference, e.Err)
+}
+
+// Unwrap implements the errors.Unwrap interface for BroadcastingError
+func (e *BroadcastingError) Unwrap() error {
+	return e.Err
+}
+
+// Is implements the errors.Is interface for BroadcastingError
+func (e *BroadcastingError) Is(target error) bool {
+	if target == nil {
+		return false
+	}
+
+	if _, ok := target.(*BroadcastingError); ok {
+		return true
+	}
+
+	if e.Err != nil {
+		return errors.Is(e.Err, target)
+	}
+
+	return false
+}
+
+// EnhanceWithCreateActionContext enhances the error with context specific to a CreateAction operation
+func EnhanceWithCreateActionContext(
+	err error,
+	txID, reference string,
+	tx []byte,
+	noSendChange []wdk.OutPoint,
+	processResult *wdk.ProcessActionResult,
+) error {
+	var broadcastErr *BroadcastingError
+	if errors.As(err, &broadcastErr) {
+		return broadcastErr.
+			WithOperation(CreateAction).
+			WithReference(reference).
+			WithTxIDIfEmpty(txID).
+			WithProcessActionContext(processResult, txID, reference).
+			WithTransactionData(tx, noSendChange)
+	}
+
+	return NewBroadcastingError(err, CreateAction).
+		WithTxID(txID).
+		WithReference(reference).
+		WithTransactionData(tx, noSendChange).
+		WithProcessActionContext(processResult, txID, reference)
+}
+
+// WithContext adds context values from processAction result
+func (e *BroadcastingError) WithContext(processResult *wdk.ProcessActionResult, txID, reference string) *BroadcastingError {
+	e.TxID = txID
+	e.Reference = reference
+	if processResult != nil {
+		e.SendWithResults = processResult.SendWithResults
+		e.ReviewResults = processResult.NotDelayedResults
+	}
+	return e
+}
+
+// WithTxID is a convenience method to set the TxID and return the error for chaining
+func (e *BroadcastingError) WithTxID(txID string) *BroadcastingError {
+	e.TxID = txID
+	return e
+}
+
+// WithPrimaryTxID sets the TxID to the first transaction ID in the slice if available
+func (e *BroadcastingError) WithPrimaryTxID(txIDs []string) *BroadcastingError {
```

Comment:
I don't really get the intention of this method, nor it gives a real context to the error.
If we really want this error in that place where it is called, then you probably should move its creation deeper where there is only single transaction.